pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -61,15 +80,13 @@ public void validate(TokenCredentials credentials) {
 
         try {
             // Parse the token
-            JWT jwt = JWTParser.parse(token);
+            final JWT jwt = JWTParser.parse(token);
 
             if (jwt instanceof SignedJWT) {
                 signedJWT = (SignedJWT) jwt;
             } else if (jwt instanceof EncryptedJWT) {
-                JWEObject jweObject = (JWEObject) jwt;
-
-                // Decrypt with shared key
-                jweObject.decrypt(new DirectDecrypter(this.secret.getBytes("UTF-8")));
+                final JWEObject jweObject = (JWEObject) jwt;
```

Comment:
In fact, I meant here, just on this line, because we should only check the existence of the `encryptionSecret` if we need to use it. Sorry if I was not clear...